pandas/tests/arrays/test_datetimelike.py

```diff
@@ -302,11 +303,26 @@ def test_searchsorted_castable_strings(self, arr1d, box):
         expected = np.array([1, 2], dtype=np.intp)
         tm.assert_numpy_array_equal(result, expected)
 
-        with pytest.raises(TypeError):
-            arr.searchsorted("foo")
-
-        with pytest.raises(TypeError):
-            arr.searchsorted([str(arr[1]), "baz"])
+        string = "foo"
+        with pytest.raises(
+            TypeError,
+            match=(
+                f"value should be a '{arr1d._scalar_type.__name__}', 'NaT', "
+                f"or array of those. Got '{type(string).__name__}' instead."
```

Comment:
`match` takes a regular expression, which is checked using `re.search`. So the `.`s will match any character - can you use `match=re.escape(` instead of `match=`?